src/classes/pilot/Pilot.ts

```diff
@@ -575,14 +575,13 @@ class Pilot implements ICloudSyncable {
   }
 
   public CanAddSkill(skill: Skill | CustomSkill): boolean {
-    if (this._level === 0) {
-      return this._skills.length < Rules.MinimumPilotSkills && !this.has('Skill', skill.ID)
-    } else {
-      const underLimit = this.CurrentSkillPoints < this.MaxSkillPoints
-      if (!this.has('Skill', skill.ID) && underLimit) return true
-      const pSkill = this._skills.find(x => x.Skill.ID === skill.ID)
-      return underLimit && pSkill && pSkill.Rank < Rules.MaxTriggerRank
-    }
+    const hasMinSkills = this._skills.length >= Rules.MinimumPilotSkills
+    return this.IsMissingSkills && (
+      !this.has('Skill', skill.ID) || (
+        hasMinSkills && 
+        this._skills.find(x => x.Skill.ID === skill.ID).Rank < Rules.MaxTriggerRank
+      )
+    )
```

Comment:
Tried to reduce some of the boolean/if/elses here to boil the statement down into a single boolean.  Originally extracted `!this.has('Skill', skill.ID)` and `this._skills.find(x => x.Skill.ID === skill.ID).Rank < Rules.MaxTriggerRank` to external `const`s, but moved the calculations to deeper in the boolean in an effort to save a small amount processing time.

I admit I'm working off of Python knowledge here, where if a boolean terminates early (e.g. if `this.IsMissingSkills` evaluates to `false`, cutting off everything after the `&&`).  If that doesn't hold true in Typescript (or is ultimately negligible), I'd prefer to extract the calculations into well-named external `const`s for clarity's sake.

Reply:
You're correct here, it'll bail as soon as hasMinSkills resolves to `false` (or null or undefined)